### PR TITLE
More specific type for 'port' component of 'parse_url()'

### PR DESCRIPTION
**Pull request by thg2k**
The `port` component can only be in the range 0-65535:

```php
<?php
var_dump(parse_url("http://localhost:-1/"));    // false
var_dump(parse_url("http://localhost:0/"));     // [ ... ]
var_dump(parse_url("http://localhost:65535/")); // [ ... ]
var_dump(parse_url("http://localhost:65536/")); // false
```

### CODE REVIEW

File: src/Type/Php/ParseUrlFunctionDynamicReturnTypeExtension.php [hunk 1]
```diff
@@ -122,17 +122,17 @@ private function cacheReturnTypes(): void
 		}
 
 		$string = new StringType();
-		$integer = new IntegerType();
+		$port = new IntegerRangeType(0, 65535);
```
Comment (staabm) [inline src/Type/Php/ParseUrlFunctionDynamicReturnTypeExtension.php]: should use `IntegerRangeType::fromInterval(...)`

Comment (herndlm) [reply src/Type/Php/ParseUrlFunctionDynamicReturnTypeExtension.php]: interesting, brb opening a PR that prevents that :)

Comment (thg2k) [reply src/Type/Php/ParseUrlFunctionDynamicReturnTypeExtension.php]: I would've never guessed that! Force-pushed with the requested change.

**Comment by thg2k** [comment]
I'm sorry I just realized I missed a couple of tests, gotta love that CI.

Made another force push to fix that, let me know if there are other outstanding issues.

**Comment by ondrejmirtes** [comment]
> gotta love that CI.

Yeah, I'm super-proud of that, thanks :)

And thanks for this PR!

**Pull request merged**